src/main/kotlin/pl/newbies/event/domain/service/EventService.kt

```diff
@@ -149,6 +124,35 @@ class EventService {
         )
     }
 
+    private fun EventDAO.appendRequestFields(request: EventRequest) {
+        val title = request.title.trim()
+
+        this.title = title
+        this.vanityUrl = getEventVanityUrl(title)
+        this.subtitle = request.subtitle?.trim()
+        request.timeFrame.let { frame ->
+            this.startDate = frame.startDate
+            this.finishDate = frame.finishDate
+        }
+        request.address.let { address ->
+            this.city = address?.city?.trim()
+            this.place = address?.place?.trim()
+            address?.coordinates.let { coordinates ->
+                this.latitude = coordinates?.latitude
+                this.longitude = coordinates?.longitude
+            }
+        }
+    }
+
+    private fun getEventVanityUrl(title: String): String =
+        StringUtils.stripAccents(title)
+            .filter { it.isLetterOrDigit() || it in listOf(' ') }
+            .replace(Regex("[ \\t]{2,}"), " ") // multiple spaces to one space
+            .trim()
+            .replace(' ', '-')
```

Comment:
why not to replace `/[ ]+/` with `-` right away?

Reply:
Oh, I can pass what characters to trim inside the `trim()` function. I'll change that